src/include/executor/execParallel.h

```diff
@@ -48,4 +48,10 @@ extern void ExecParallelReinitialize(PlanState *planstate,
 
 extern void ParallelQueryMain(dsm_segment *seg, shm_toc *toc);
 
+typedef void (*ParallelQueryMain_hook_type)(shm_toc *toc);
+extern PGDLLIMPORT ParallelQueryMain_hook_type ParallelQueryMain_hook;
+
+typedef void (*ExecInitParallelPlan_hook_type)(EState *estate, ParallelContext *pcxt, bool estimate);
```

Comment:
1. What's the meaning of the third argument "estimate"?
2. Why do we need to call this function twice in the engine with `estimate = true/false`?

Please add a comment that can answer those two questions. It should be understandable by readers without looking at Babelfish implementation.

Reply:
Added explaination. We first need to estimate the memory requirement before we can actually send data to parallel worker